pkg/tbtc/node.go

```diff
@@ -470,8 +470,81 @@ func (n *node) handleRedemptionProposal(
 	startBlock uint64,
 	delayBlocks uint64,
 ) {
-	// TODO: Implementation.
-	panic("not implemented yet")
+	wallet, ok := n.walletRegistry.getWalletByPublicKeyHash(
```

Comment:
I am afraid we may have the same problems as in sweeping with unregistered unmarshalers: https://github.com/keep-network/keep-core/issues/3611.

I don't think it's blocking this PR but it's something we should address both for sweeps and redemptions before doing the user acceptance tests.

Reply:
Yeah, sure!